### PR TITLE
Housekeeping: group lookup data together, store in an optional struct

### PR DESCRIPTION
**Pull request by mrmr1993**
This PR groups the lookup information in the constraint system and verifier index together in an optional structure, removing a TODO.

### CODE REVIEW

File: circuits/kimchi/src/nolookup/constraints.rs [hunk 1]
```diff
@@ -171,21 +198,8 @@ pub struct ConstraintSystem<F: FftField> {
...
-    pub lookup_tables8: Vec<Vec<E<F, D<F>>>>,
-
-    /// Lookup selectors:
-    /// For each kind of lookup-pattern, we have a selector that's
-    /// 1 at the rows where that pattern should be enforced, and 0 at
-    /// all other rows.
-    #[serde_as(as = "o1_utils::serialization::SerdeAs")]
-    pub lookup_selectors: Vec<E<F, D<F>>>,
+    /// lookup constraint system
+    pub lookup_constraint_system: Option<LookupConstraintSystem<F>>,
```
Comment (mimoo) [inline circuits/kimchi/src/nolookup/constraints.rs:202]: omg this is awesome <3

File: circuits/kimchi/src/nolookup/constraints.rs [hunk 2]
```diff
@@ -31,6 +31,33 @@ pub const ZK_ROWS: u64 = 3;
 // ConstraintSystem
 //
 
+#[serde_as]
+#[derive(Clone, Serialize, Deserialize, Debug)]
+pub struct LookupConstraintSystem<F: FftField> {
```
Comment (mimoo) [inline circuits/kimchi/src/nolookup/constraints.rs:36]: should this live in `lookup.rs`?

Comment (mrmr1993) [reply circuits/kimchi/src/nolookup/constraints.rs:36]: It's a part of the constraint system, so I'm inclined to say it should stay here, unless we're planning to split out all of the polys for the other gates too.

Comment (mimoo) [reply circuits/kimchi/src/nolookup/constraints.rs:36]: (btw this struct doesn't have documentation)

**Review by mimoo — APPROVED** [review]
Nice cleanup :)

File: dlog/kimchi/src/bench.rs [hunk 3]
```diff
@@ -0,0 +1,116 @@
+use crate::{index::Index, prover::ProverProof};
```
Comment (mimoo) [inline dlog/kimchi/src/bench.rs:1]: merge issue: delete this file

**Comment by mimoo** [comment]
LGTM besides the new file that was added

**Pull request merged**